packages/sol-cov/src/coverage_manager.ts

```diff
@@ -131,18 +112,53 @@ export class CoverageManager {
         };
         return partialCoverage;
     }
+    constructor(
+        artifactAdapter: AbstractArtifactAdapter,
+        getContractCodeAsync: (address: string) => Promise<string>,
+        verbose: boolean,
+    ) {
+        this._getContractCodeAsync = getContractCodeAsync;
+        this._artifactAdapter = artifactAdapter;
+        this._verbose = verbose;
+    }
+    public appendTraceInfo(traceInfo: TraceInfo): void {
+        // console.log(JSON.stringify(traceInfo, null, '\n'));
+        this._traceInfos.push(traceInfo);
+    }
+    public async writeCoverageAsync(): Promise<void> {
+        const finalCoverage = await this._computeCoverageAsync();
+        const stringifiedCoverage = JSON.stringify(finalCoverage, null, '\t');
+        await mkdirpAsync('coverage');
+        fs.writeFileSync('coverage/coverage.json', stringifiedCoverage);
+    }
     private async _computeCoverageAsync(): Promise<Coverage> {
+        const contractsData = await this._artifactAdapter.collectContractsDataAsync();
         const collector = new Collector();
         for (const traceInfo of this._traceInfos) {
             if (traceInfo.address !== constants.NEW_CONTRACT) {
                 // Runtime transaction
                 let runtimeBytecode = (traceInfo as TraceInfoExistingContract).runtimeBytecode;
                 runtimeBytecode = addHexPrefix(runtimeBytecode);
-                const contractData = _.find(this._contractsData, { runtimeBytecode }) as ContractData;
+                const contractData = _.find(contractsData, contractDataCandidate => {
+                    // Library linking placeholder: __ConvertLib____________________________
+                    let runtimeBytecodeRegex = contractDataCandidate.runtimeBytecode.replace(/_.*_/, '.*');
+                    // Last 86 characters is solidity compiler metadata that's different between compilations
+                    runtimeBytecodeRegex = runtimeBytecodeRegex.replace(/.{86}$/, '');
+                    // Libraries contain their own address at the beginning of the code and it's impossible to know it in advance
+                    runtimeBytecodeRegex = runtimeBytecodeRegex.replace(
+                        /^0x730000000000000000000000000000000000000000/,
+                        '0x73........................................',
+                    );
+                    return !_.isNull(runtimeBytecode.match(runtimeBytecodeRegex));
+                }) as ContractData;
                 if (_.isUndefined(contractData)) {
-                    throw new Error(`Transaction to an unknown address: ${traceInfo.address}`);
+                    if (this._verbose) {
+                        // tslint:disable-next-line:no-console
+                        console.warn(`Transaction to an unknown address: ${traceInfo.address}`);
```

Comment:
I'd prefer we make a `verbose` module/class. And have a method called `verbose.warn('...message...')` and it internally checks if verbosity is turned on or not. This means we don't need this conditional check everywhere we want to log a verbose warning.